crates/turbopack-core/src/source_map.rs

```diff
@@ -82,80 +82,84 @@ impl<'a> From<sourcemap::Token<'a>> for Token {
     }
 }
 
-impl SourceMap {
-    /// Encoding a SourceMap stringifies it into JSON.
-    #[async_recursion]
-    async fn encode<W: Write + Send>(&self, w: &mut W) -> Result<()> {
-        match self {
-            SourceMap::Regular(r) => r.0.to_writer(w)?,
+impl SourceMapVc {
+    /// Creates a new SourceMap::Regular Vc out of a sourcemap::SourceMap
+    /// ("CrateMap") instance.
+    pub fn new_regular(map: CrateMap) -> Self {
+        SourceMap::Regular(RegularSourceMap::new(map)).cell()
+    }
+
+    /// Creates a new SourceMap::Sectioned Vc out of a collection of source map
+    /// sections.
+    pub fn new_sectioned(sections: Vec<SourceMapSection>) -> Self {
+        SourceMap::Sectioned(SectionedSourceMap::new(sections)).cell()
+    }
+}
+
+#[turbo_tasks::value_impl]
+impl SourceMapVc {
+    /// Stringifies the source map into JSON bytes.
+    #[turbo_tasks::function]
+    pub async fn to_rope(self) -> Result<RopeVc> {
+        let this = self.await?;
+        let rope = match &*this {
+            SourceMap::Regular(r) => {
+                let mut bytes = vec![];
+                r.0.to_writer(&mut bytes)?;
+                Rope::from(bytes)
+            }
 
             SourceMap::Sectioned(s) => {
+                if s.sections.len() == 1 {
+                    let s = &s.sections[0];
+                    if s.offset == (0, 0) {
+                        return Ok(s.map.to_rope());
+                    }
+                }
+
                 // My kingdom for a decent dedent macro with interpolation!
-                write!(
-                    w,
-                    r#"{{
+                let mut rope = RopeBuilder::from(
```

Comment:
Unrelated to this PR, but have we considered solutions from the ecosystem like https://github.com/dtolnay/indoc?